src/app/parcels/ParcelsPage.tsx

```diff
@@ -482,12 +484,17 @@ const ParcelsPage: React.FC<{}> = () => {
                                     <ExpandedParcelDetails parcelId={selectedParcelId} />
                                 </Suspense>
                             </ContentDiv>
-
                             <ButtonsDiv>
                                 <Centerer>
                                     <LinkButton link={`/parcels/edit/${selectedParcelId}`}>
                                         Edit Parcel
                                     </LinkButton>
+                                    <LinkButton link={`/clients?clientId=${selectedPrimaryKey}`}>
```

Comment:
Could you explain how `selectedPrimaryKey` is set when we directly go to that modal page `.../pacels/pacelId=...`? I don't think it's propertly set. To demonstrate this, tru
- opening the modal by clicking on a row
- copy the URL
- open that URL in a new tab
- click "See client"

Reply:
I see what you mean now, I've amended the useEffect to set the primaryKey and I can access the client page now even if I  copy the URL to another tab.